File: shared/types/file.ts
/**
 * File-related type definitions shared between frontend and backend
 */

/**
 * File model
 */
export interface File {
  id: string;
  projectId: string;
  name: string;
  path: string;
  type: FileType;
  mimeType: string;
  size: number;
  createdAt: string; // ISO date string
  updatedAt: string; // ISO date string
  createdBy: string;
  url: string;
  metadata: Record<string, any>;
}

/**
 * File type
 */
export type FileType = 'document' | 'image' | 'spreadsheet' | 'other';

/**
 * File upload response
 */
export interface FileUploadResponse {
  file: File;
  uploadUrl: string;
}

/**
 * File download response
 */
export interface FileDownloadResponse {
  file: File;
  downloadUrl: string;
}

/**
 * File update request
 */
export interface UpdateFileRequest {
  name?: string;
  metadata?: Record<string, any>;
} 

File: services/file-service/app/models/file.py
from pydantic import BaseModel, Field
from typing import Dict, Any, Optional, Literal
from datetime import datetime

# File type enum
FileType = Literal["document", "image", "spreadsheet", "other"]

class FileBase(BaseModel):
    """Base file model with common attributes"""
    name: str
    projectId: str
    type: FileType = "other"
    metadata: Dict[str, Any] = Field(default_factory=dict)

class FileCreate(FileBase):
    """Model for file creation"""
    path: Optional[str] = None
    mimeType: str
    size: int
    createdBy: str

class FileUpdate(BaseModel):
    """Model for file update"""
    name: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None

class FileInDB(FileBase):
    """File model as stored in the database"""
    id: str
    path: str
    mimeType: str
    size: int
    createdAt: datetime
    updatedAt: datetime
    createdBy: str
    url: Optional[str] = None

class File(FileInDB):
    """File model as returned to clients"""
    createdAt: str  # ISO date string
    updatedAt: str  # ISO date string

class FileUploadResponse(BaseModel):
    """Response model for file upload request"""
    file: File
    uploadUrl: str

class FileDownloadResponse(BaseModel):
    """Response model for file download request"""
    file: File
    downloadUrl: str 